### ISSUE TITLE
[BUG] 1D scattering output doesn't seem to garbage collect

### ISSUE THREAD

**Issue by eickenberg** [kymatio/kymatio#400]
I have a large dataset to process, and it seems like I can't get the results of `S1D(x)`  to leave the GPU, so eventually the memory overflows no matter how small the batch was. This needs to be fixed

**Comment by eickenberg** [comment]
I did another user error, where during manual batching I combined two techniques for strided iteration which led to bigger and bigger batches.

**Comment by eickenberg** [comment]
Actually the bug persists: My hunch is that it happens (probably) only when we do not vectorize the output (`vectorize=False` is a necessary setting for `average=False`). Maybe the garbage collector gets confused

**Comment by eickenberg** [comment]
The bug was in my code to detach and cpu all items in a dictionary. I had written this:

```python
def detach_dict(d):
    output = {}
    for k, v in d.items():
        try:
            dv = v.detach().cpu().numpy()
        except:
            dv = v
            print("couldn't detach because that probably isn't a pytorch tensor")
        output[k] = dv
    return output
```

and had set `output[k] = v` instead of `dv`
